### ISSUE TITLE
Feature: Ability to cache CHD files instead of ISO when extracting zip files (say for PS2 iso or bin/cue games)

### ISSUE THREAD

**Issue by geogolem** [fraganator/archive-cache-manager#45]
So this may be a little complicated - but let me try and explain what I would like.

Let's first give some context to the problem.
Imagine I have a PS2 romset that I validate/maintain using romvault and the redump dat. This romset will consist of zip files, that contain ISO's (or in some cases bin/cue files). The PS2 emulator cannot play zip files... so ticking the option to extract prior to launching the game is necessary.. and then this amazing archive-cache-manager plugin comes in handy because then the file can be cached so that subsequent launches of the already extracted/cached game can be much faster.

All this is great.. but the PS2 emulator is also able to play CHD files directly.... CHD files take up way less space that the unzipped iso files...

I like to keep the original zipped files so that I can maintain/update my Ps2 rom collection as changes are made to the NoIntro dat... so I don't want to just get rid of my original zipped romset...

What I would like is this --> Import the zipped PS2 roms via Launchbox --> whenever I go to launch a PS2 game have this plugin extract the ISO from the zip file and then convert it to CHD using latest chdman and store it in the cache.... Then the CHD can be launched in the emulator.

I could then mark the CHD cached file as "keep"..

The idea is that, I would be storing my PS2 games in zip files to match the nointro dat.. but I could easily create CHD versions of all the games or the games i play often etc..

I realize, the archive-cache-manager plugin as the ability to extract CHD files prior to launch.. but basically I'm requesting to store the cached extracted files in CHD format instead of ISO etc. Does this make sense?

**Comment by geogolem** [comment]
Just wanted to provide some more info for clarity...

Basically if I want to keep my PS2 roms in zip format corresponding to redump dat and maintained using romvault... I need to do 1 of 2 things.

1) import the roms without copying/moving using launchbox, and use this plugin to extract and cache the ISO (or bin/cue etc.) prior to launching.. or 

2) manually extract all the zip files to another folder, import all these unzipped roms into Launchbox.. then use a tool like: https://forums.launchbox-app.com/files/file/3697-bulk-convert-cuegdiiso-files-to-chd/
in order to convert all these roms to CHD... 

Basically I would like to somehow accomplish (2) using (1) so that I can sort of build/create a CHD set on demand... This would work nice too, because if my noIntro romset has some additions, the games can be added to launchbox and CHD's generated upon first launch of said game.. etc.

Not sure if this is overly complicated, but hopefully it can illustrate the motivation for the feature request.

**Comment by geogolem** [comment]
Also wanted to mention.. maybe a better option would be the ability to manually add cached files..

For example, I could then manually extract some iso's from zip files.. manually convert them into CHD format.. and then the archive-cache-manager plugin could have the ability for me to import/specify a file to use as a cached file for a game.. Then it can copy/move the CHD file instead of extracting/converting if one already has the CHD file available...

----

or maybe the conversion from cached iso to chd can be a separate step in the plugin configuration where you can convert the iso to CHD...
